Lib/graphlib.py

```diff
@@ -248,3 +248,78 @@ def static_order(self):
             self.done(*node_group)
 
     __class_getitem__ = classmethod(GenericAlias)
+
+
+def reverse(graph):
+    """Reverse the direction of the edges in a directed graph.
+
+    Given a mapping from nodes to collections of their dependencies,
+    construct a dict mapping each node to the set of nodes that depend on it.
+
+    Nodes that have no dependents appear as keys in the result, with an empty
+    set as value.
+
+    For example:
+
+        >>> reverse({"a": ["b", "c"], "d": []})
+        {'b': {'a'}, 'c': {'a'}, 'a': set(), 'd': set()}
+
+    """
+    result = {}
+    for node, deps in graph.items():
+        for dep in deps:
+            if dep not in result:
+                result[dep] = {node}
+            else:
+                result[dep].add(node)
+        if node not in result:
+            result[node] = set()
+    return result
+
+
+def as_transitive(graph):
+    """Compute the transitive closure of a dependency graph.
+
+    If the input graph contains cycles, raise CycleError.
```

Comment:
```suggestion
    If the input graph contains cycles, raise a CycleError.
```

Reply:
This doesn't need an article. A quick `rg` suggests both without and with an article are common, e.g. without

```
Lib/urllib/request.py
627:        perform the redirect.  Otherwise, raise HTTPError if no-one

Lib/asyncio/tasks.py
422:    Note: This does not raise TimeoutError! Futures that aren't done

Lib/json/__init__.py
154:    of obj or raise TypeError. The default simply raises TypeError.
```

vs with

```
Lib/json/encoder.py
144:        version of the object or raise a ``TypeError``.

Lib/unittest/mock.py
1262:      any attribute not in this list will raise an `AttributeError`.
```

Reply:
Try to grep with `:exc:` We generally use the role when talking about exceptions being raised. I personally prefer having an article.